### DISCUSSION TITLE
Preview files in ".doc" and ".docx" formats

### DISCUSSION THREAD

**Discussion by Limaferreira** [sepinf-inc/IPED#851] [General]
Good night,

I am unable to preview ".doc" and ".docx" files on the IPED system. There is some specific setting to enable this function. Other file formats such as: .pdf .jpeg .png .xls and .xlsx I am able to view normally.

**Comment by lfcnassif** [comment]
It should work. What is your operating system, java version, iped version? Please send the Iped-SearchApp.log from your case.

**Reply by Limaferreira** [reply]
The operating system is win 10, java version is 8 update 311 and IPED version is 3.18.9.

**Reply by lfcnassif** [reply]
> Please send the Iped-SearchApp.log from your case.

This is needed, together with a sample triggering file to reproduce this, so we could confirm the issue and test a possible fix.

**Reply by Limaferreira** [reply]
Follows the message that is shown in Iped-SearchApp.log when clicking on a file in .doc or docx format: 



Killed pid 57080 exitValue: 0
2021-11-29 14:23:20	[INFO]	[application.connection.LocalOfficeConnection]			Opening local OpenOffice.org connection.
2021-11-29 14:23:21	[INFO]	[fileViewer.frames.LibreOfficeViewer]			LibreOffice running.
2021-11-29 14:23:21	[INFO]	[fileViewer.frames.LibreOfficeViewer$1]			Constructing LibreOffice frame...
2021-11-29 14:23:22	[INFO]	[fileViewer.frames.LibreOfficeViewer$1]			LibreOffice frame ok
2021-11-29 14:23:22	[INFO]	[fileViewer.frames.LibreOfficeViewer]			LibreOffice restarted.
2021-11-29 14:23:26	[INFO]	[indexer.desktop.FileProcessor]			Opening /IMAGEM.E01/vol_vol6/pagefile.sys>>Carved-3499039152.doc
2021-11-29 14:23:26	[INFO]	[fileViewer.frames.LibreOfficeViewer$2]			ag.ion.bion.officelayer.document.DocumentException: Unsupported URL <file:////Espelhamento%20de%20Apreensões/Notebook%20Positivo/IPED/tempIPED/indexador-temp1638206576320/iped7780466308159678320.doc>: "type detection failed"
2021-11-29 14:23:26	[INFO]	[fileViewer.frames.LibreOfficeViewer]			Restarting LibreOffice...
Killed pid 52732 exitValue: 0

**Reply by lfcnassif** [reply]
Could you test removing all spaces and diacritics (accents) from your case path and restarting the case? What is the full path of `Espelhamento%20de%20Apreensões/Notebook%20Positivo/IPED` folder (the path seems malformed)? Do you have write permission on it?

**Reply by lfcnassif** [reply]
More questions: are you able to open those files double clicking on them, using an external application? This happens with every doc/docx file or just with some of them? If last option, we really need a sample file to reproduce this.

**Reply by Limaferreira** [reply]
> Could you test removing all spaces and diacritics (accents) from your case path and restarting the case? What is the full path of Espelhamento%20de%20Apreensões/Notebook%20Positivo/IPED folder (the path seems malformed)? Do you have write permission on it?

So the files were indexed in a folder on the network, so the link is long. As mentioned before, other formats like: JPEG, PDF and XLS the preview works normally, only .doc .docx and .pptx are not showing the preview. I indexed and I'm accessing with the admin user, I have full permission on the network folders

**Reply by Limaferreira** [reply]
> More questions: are you able to open those files double clicking on them, using an external application? This happens with every doc/docx file or just with some of them? If last option, we really need a sample file to reproduce this.


Double-clicking files open normally.

**Reply by lfcnassif** [reply]
> So the files were indexed in a folder on the network

This is a very important info.

> Unsupported URL file:////Espelhamento%20de%20Apreensões/Notebook%20Positivo/IPED/tempIPED/indexador-temp1638206576320/iped7780466308159678320.doc: "type detection failed"

Looking at this more carefully, seems your indexTemp folder is configured in a network share, what I **strongly do not recommend**. I was able to reproduce this here pointing the iped temp to an unmapped network folder. Mapping the network share and configuring the drive letter accordingly in `indexTemp` in [your_case]/indexador/LocalConfig.txt was enough to make it work, try it. But I recommend you configuring `indexTemp` in a local drive (SSD if possible) for all cases.

**Reply by Limaferreira** [reply]
OK! I will configure the "indexTemp" folder locally and run the tests again. Thank you for the informations.
